File: 2024-XXC-Part-2/Day10-No-Modifier/main.php

<?php

include 'src/parse.php';
include 'src/scoreMap.php';
include 'src/scoreTrail.php';
include 'src/getTrails.php';

error_reporting(E_ERROR | E_PARSE);
echo 'Part 1: What is the sum of the scores of all trailheads on your topographic map? Answer: ' .
        scoreMap(
            parseInput(
                // file_get_contents('challenge_input.dat')
                    file_get_contents('test_input.dat')
            )
    ) . PHP_EOL;

File: 2024-XXC-Part-2/Day10-No-Modifier/haskell/app/Main.hs
module Main where

import SolvePart1 (solvePart1)
import SolvePart2 (solvePart2)

main :: IO ()
main = do
  input <- readFile "data/challenge_input.dat"
  putStrLn "Part 1: What is the sum of the scores of all trailheads on your topographic map?"
  putStrLn $ solvePart1 input
  putStrLn "Part 2: What is the sum of the ratings of all trailheads?"
  putStrLn $ solvePart2 input
